### PR TITLE
feat(core): add vertex-anthropic engine for Vertex-hosted Claude models

### PR DESCRIPTION
**Pull request by djdanielsson**
## Summary

- **Add `vertex-anthropic` engine** backed by `@ai-sdk/google-vertex/anthropic` for Anthropic Claude models hosted on Google Vertex AI or corporate Vertex-compatible proxies.
- **Support Bearer-token proxy authentication**: when an API key is configured, it is injected as the Authorization header via a synthetic `authClient` passed to `googleAuthOptions`, bypassing Google credential discovery. When no key is configured, standard Google Cloud ADC is used.
- **Custom fetch wrapper** handles proxy compatibility: strips `stream_options`/`stream` fields the Anthropic Vertex API rejects, and converts JSON responses to SSE event streams when the proxy returns `application/json` instead of `text/event-stream`. For proxies with self-signed certificates, the standard `NODE_EXTRA_CA_CERTS` environment variable is recommended.
- **Add `--skip-extension` build flag** to skip VSIX packaging in container builds where the VS Code extension is unnecessary.

## Commits

- `chore(build): add --skip-extension flag to skip VSIX packaging` -- Container builds don't need the VS Code extension; allow skipping with `--skip-extension` or `ABBENAY_SKIP_EXTENSION=1` env var. Stage 6 (`--code-install`) is also skipped when extension packaging is skipped. Used in the Containerfile.
- `feat(core): add vertex-anthropic engine for Vertex-hosted Claude models` -- New engine entry in the data-driven registry, `@ai-sdk/google-vertex` dependency, custom fetch wrapper for proxy compat, proxy-compat helpers extracted and tested (13 tests), documentation updates (CONFIGURATION.md, CORE.md), and DR-026 decision log entry.

### Configuration example (Bearer-token proxy)

```yaml
providers:
  corp-vertex:
    engine: vertex-anthropic
    base_url: "https://your-proxy.example.com/sonnet/models"
    api_key_env_var_name: "VERTEX_ANTHROPIC_API_KEY"
    models:
      claude-sonnet-4:
        model_id: "claude-sonnet-4@20250514"
```

The `base_url` carries the full URL prefix including any proxy-specific path segments; the SDK appends `/<model-id>:streamRawPredict` automatically.

For proxies with self-signed or internal CA certificates, set `NODE_EXTRA_CA_CERTS=/path/to/ca-bundle.crt`.

### Configuration example (standard Google Cloud ADC)

```yaml
providers:
  vertex-claude:
    engine: vertex-anthropic
    models:
      claude-sonnet-4:
        model_id: "claude-sonnet-4@20250514"
```

## Files changed

| File | Change |
|------|--------|
| `packages/daemon/src/core/engines.ts` | New `vertex-anthropic` engine, `vertexAnthropicProvider()` factory with synthetic authClient, custom fetch wrapper for body sanitization and JSON-to-SSE conversion |
| `packages/daemon/src/core/engines.test.ts` | 13 new tests for `sanitizeVertexRequestBody` and `convertAnthropicJsonToSse` helpers |
| `packages/daemon/package.json` | Add `@ai-sdk/google-vertex` dependency |
| `packages/daemon/build.js` | Add `@ai-sdk/google-vertex` to `@abbenay/core` peer dependencies |
| `package-lock.json` | Lockfile update |
| `build.js` | Add `--skip-extension` / `ABBENAY_SKIP_EXTENSION` flag |
| `Containerfile` | Set `ABBENAY_SKIP_EXTENSION=1` in container build |
| `.audit-allowlist` | Add advisory for serialize-javascript (dev-only, existing dep chain) |
| `docs/CONFIGURATION.md` | Add `vertex-anthropic` to engines table, usage examples, and `NODE_EXTRA_CA_CERTS` guidance |
| `docs/CORE.md` | Add `vertex-anthropic` to engines table and install instructions |
| `docs/decisions.md` | DR-026: Vertex Anthropic engine with Bearer-token proxy support |

## Test plan

- [x] `npm run lint` passes locally (0 errors)
- [x] `npm test` passes locally (19 test files, 420 tests)
- [x] `npm run ci:build` passes locally
- [x] `sanitizeVertexRequestBody` tested: strips fields, no-op when clean, handles invalid JSON
- [x] `convertAnthropicJsonToSse` tested: text-only conversion, tool_use rejection, metadata preservation, stop_reason propagation, multi-block concatenation, malformed response shapes
- [x] Existing `engines.test.ts` invariant tests validate `PROVIDER_LOADERS` stays in sync
- [x] Docs updated for user-facing changes (CONFIGURATION.md, CORE.md)
- [x] Architectural decision recorded (DR-026)

### CODE REVIEW

File: packages/daemon/src/core/engines.ts [hunk 1]
```diff
@@ -139,6 +147,102 @@ async function openaiCompatibleProvider(
...
+    ? new Agent({ connect: { rejectUnauthorized: false } })
+    : undefined;
+  if (insecureTls) {
+    console.warn('[Adapter] vertex-anthropic running with ABBENAY_INSECURE_TLS enabled');
+  }
+  providerConfig.fetch = (input: string | URL | Request, init?: RequestInit) => {
+    const patchedInit = { ...(init ?? {}) } as Record<string, unknown>;
+    if (typeof patchedInit.body === 'string') {
+      try {
+        const body = JSON.parse(patchedInit.body as string) as Record<string, unknown>;
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Added 11 focused tests for sanitizeVertexRequestBody and convertAnthropicJsonToSse

File: packages/daemon/package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "@ai-sdk/deepseek": "^2.0.0",
     "@ai-sdk/fireworks": "^2.0.0",
     "@ai-sdk/google": "^3.0.0",
+    "@ai-sdk/google-vertex": "^4.0.102",
     "@ai-sdk/groq": "^3.0.0",
     "@ai-sdk/mcp": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/package.json:33]: Removed undici import entirely (no longer needed)

File: packages/daemon/src/core/engines.ts [hunk 3]
```diff
@@ -139,6 +147,102 @@ async function openaiCompatibleProvider(
...
+    };
+  }
+  // Custom fetch wrapper: strips fields the Anthropic Vertex API rejects
+  // (e.g. stream_options) and optionally disables TLS verification for
+  // corp/internal proxies with self-signed certificate chains.
+  const insecureTls = _isTruthyEnv(process.env.ABBENAY_INSECURE_TLS);
+  const insecureAgent = insecureTls
+    ? new Agent({ connect: { rejectUnauthorized: false } })
+    : undefined;
+  if (insecureTls) {
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Removed the flag completely; documented NODE_EXTRA_CA_CERTS as the recommended approach

File: packages/daemon/src/core/engines.ts [hunk 4]
```diff
@@ -139,6 +147,102 @@ async function openaiCompatibleProvider(
...
+        const text = await resp.text();
+        let msg: Record<string, unknown>;
+        try { msg = JSON.parse(text) as Record<string, unknown>; } catch { return new Response(text, { status: resp.status, statusText: resp.statusText, headers: resp.headers }); }
+        const content = Array.isArray(msg.content) ? msg.content : [];
+        let fullText = '';
+        for (const block of content) {
+          if (block && typeof block === 'object' && (block as Record<string,unknown>).type === 'text') {
+            fullText += (block as Record<string,string>).text || '';
+          }
+        }
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Now returns null for non-text blocks, passing original response through unchanged

File: packages/daemon/src/core/engines.ts [hunk 5]
```diff
@@ -727,7 +843,12 @@ export async function* streamChat(
 
     // Iterate over the full stream and yield our ChatChunk format
     let gotContent = false;
+    let partCount = 0;
     for await (const part of result.fullStream) {
+      partCount++;
+      if (partCount <= 3) {
+        console.warn(`[Adapter] fullStream part #${partCount}: type=${part.type}${part.type === 'text-delta' ? ` text=${JSON.stringify((part as {text:string}).text).slice(0,80)}` : ''}${part.type === 'finish' ? ` reason=${(part as {finishReason:string}).finishReason}` : ''}`);
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Removed the debug logging entirely (it was temporary instrumentation)

File: packages/daemon/src/core/engines.ts [hunk 6]
```diff
@@ -139,6 +147,102 @@ async function openaiCompatibleProvider(
...
+        const body = JSON.parse(patchedInit.body as string) as Record<string, unknown>;
+        const removed: string[] = [];
+        if ('stream_options' in body) { delete body.stream_options; removed.push('stream_options'); }
+        if ('stream' in body) { delete body.stream; removed.push('stream'); }
+        patchedInit.body = JSON.stringify(body);
+        const url = typeof input === 'string' ? input : input instanceof URL ? input.href : input.url;
+        console.warn(`[Adapter] fetch: ${url} removed=[${removed}] body_keys=[${Object.keys(body)}]`);
+      } catch { /* non-JSON body, leave as-is */ }
+    } else {
+      console.warn(`[Adapter] fetch: body type=${typeof patchedInit.body}`);
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Switched to debug() and only logs when fields are actually removed

File: packages/daemon/src/core/engines.ts [hunk 7]
```diff
@@ -139,6 +140,132 @@ async function openaiCompatibleProvider(
...
+    } else {
+      return null;
+    }
+  }
+  const events = [
+    `event: message_start\ndata: ${JSON.stringify({ type: 'message_start', message: { id: msg.id || '', type: 'message', role: 'assistant', content: [], model: msg.model || '', stop_reason: null, stop_sequence: null, usage: msg.usage || { input_tokens: 0, output_tokens: 0 } } })}\n`,
+    `event: content_block_start\ndata: ${JSON.stringify({ type: 'content_block_start', index: 0, content_block: { type: 'text', text: '' } })}\n`,
+    `event: content_block_delta\ndata: ${JSON.stringify({ type: 'content_block_delta', index: 0, delta: { type: 'text_delta', text: fullText } })}\n`,
+    `event: content_block_stop\ndata: ${JSON.stringify({ type: 'content_block_stop', index: 0 })}\n`,
+    `event: message_delta\ndata: ${JSON.stringify({ type: 'message_delta', delta: { stop_reason: 'end_turn', stop_sequence: null }, usage: msg.usage || { output_tokens: 0 } })}\n`,
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Propagated stop_reason and stop_sequence from the parsed JSON response, falling back to 'end_turn' / null when absent. Added 2 tests covering both cases.

File: packages/daemon/src/core/engines.ts [hunk 8]
```diff
@@ -139,6 +140,132 @@ async function openaiCompatibleProvider(
...
+ * Create model via @ai-sdk/google-vertex/anthropic (dynamically loaded).
+ *
+ * Supports two modes:
+ * - Standard Vertex AI: uses Google Cloud auth (requires GOOGLE_APPLICATION_CREDENTIALS)
+ * - Bearer-token proxy: when apiKey is set, injects it as the Authorization
+ *   header via a synthetic authClient, bypassing Google credential discovery.
+ *
+ * For corporate proxies with self-signed certificates, set the standard
+ * NODE_EXTRA_CA_CERTS environment variable to the CA bundle path instead
+ * of disabling TLS verification.
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts:230]: Updated the PR description to remove all references to ABBENAY_INSECURE_TLS and document NODE_EXTRA_CA_CERTS as the recommended approach instead.

File: packages/daemon/src/core/engines.ts [hunk 9]
```diff
@@ -139,6 +140,136 @@ async function openaiCompatibleProvider(
...
+ */
+async function vertexAnthropicProvider(
+  config: ProviderFactoryConfig,
+  modelId: string,
+): Promise<LanguageModel> {
+  const factory = await loadProviderFactory(
+    '@ai-sdk/google-vertex/anthropic', 'createVertexAnthropic',
+  ) as VertexAnthropicProviderFactory;
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts:238]: Fixed. loadProviderFactory now extracts the root package name from subpath exports (e.g. @ai-sdk/google-vertex/anthropic → @ai-sdk/google-vertex) for the install hint.

File: packages/daemon/src/core/engines.ts [hunk 10]
```diff
@@ -139,6 +140,136 @@ async function openaiCompatibleProvider(
...
+    : null;
+  const events = [
+    `event: message_start\ndata: ${JSON.stringify({ type: 'message_start', message: { id: msg.id || '', type: 'message', role: 'assistant', content: [], model: msg.model || '', stop_reason: null, stop_sequence: null, usage: msg.usage || { input_tokens: 0, output_tokens: 0 } } })}\n`,
+    `event: content_block_start\ndata: ${JSON.stringify({ type: 'content_block_start', index: 0, content_block: { type: 'text', text: '' } })}\n`,
+    `event: content_block_delta\ndata: ${JSON.stringify({ type: 'content_block_delta', index: 0, delta: { type: 'text_delta', text: fullText } })}\n`,
+    `event: content_block_stop\ndata: ${JSON.stringify({ type: 'content_block_stop', index: 0 })}\n`,
+    `event: message_delta\ndata: ${JSON.stringify({ type: 'message_delta', delta: { stop_reason: stopReason, stop_sequence: stopSequence }, usage: msg.usage || { output_tokens: 0 } })}\n`,
+    `event: message_stop\ndata: ${JSON.stringify({ type: 'message_stop' })}\n`,
+  ];
+  return events.join('\n');
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Fixed. Added trailing newline so the last message_stop event is properly delimited for SSE parsers.

File: packages/daemon/src/core/engines.ts [hunk 11]
```diff
@@ -139,6 +143,136 @@ async function openaiCompatibleProvider(
...
+        const text = await resp.text();
+        const sseBody = convertAnthropicJsonToSse(text);
+        if (sseBody) {
+          debug(`[Adapter] vertex-anthropic: converted JSON response to SSE`);
+          const sseHeaders = new Headers(resp.headers);
+          sseHeaders.set('content-type', 'text/event-stream');
+          return new Response(sseBody, { status: 200, statusText: 'OK', headers: sseHeaders });
+        }
+        debug(`[Adapter] vertex-anthropic: JSON response contains non-text blocks, passing through`);
+        return new Response(text, { status: resp.status, statusText: resp.statusText, headers: resp.headers });
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts:280]: Added an explicit console.warn when non-text blocks are detected in a JSON response, explaining that tool calling requires the proxy to return text/event-stream. Also documented this limitation in CONFIGURATION.md.

File: docs/CONFIGURATION.md [hunk 12]
```diff
@@ -248,6 +249,52 @@ providers:
...
+
+```bash
+export VERTEX_ANTHROPIC_API_KEY="your-bearer-token-here"
+```
+
+The `base_url` should include the full path prefix up to `/models` (the engine
+appends `/<model-id>:streamRawPredict` automatically). When no `api_key` is
+configured, the engine falls back to standard Google Cloud Application Default
+Credentials.
+
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply docs/CONFIGURATION.md:278]: Added the suggested limitation note to CONFIGURATION.md.

File: packages/daemon/src/core/engines.ts [hunk 13]
```diff
@@ -139,6 +143,139 @@ async function openaiCompatibleProvider(
...
+        if (sseBody) {
+          debug(`[Adapter] vertex-anthropic: converted JSON response to SSE`);
+          const sseHeaders = new Headers(resp.headers);
+          sseHeaders.set('content-type', 'text/event-stream');
+          return new Response(sseBody, { status: 200, statusText: 'OK', headers: sseHeaders });
+        }
+        console.warn(
+          '[Adapter] vertex-anthropic: proxy returned application/json with non-text content blocks (e.g. tool_use). ' +
+          'JSON→SSE conversion only supports text blocks. Tool calling requires the proxy to return text/event-stream.',
+        );
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Refactored convertAnthropicJsonToSse to return a discriminated union { ok, body } | { ok, reason } with distinct 'parse-error' and 'non-text-content' reasons. The caller now logs an accurate warning for each case. Tests updated to assert specific reasons.

File: build.js [hunk 14]
```diff
@@ -43,6 +43,7 @@ const PROTO_ONLY = args.has('--proto-only');
 const SKIP_SEA = args.has('--skip-sea');
 const SKIP_PROTO = args.has('--skip-proto');
 const SKIP_ZIP = args.has('--skip-zip');
+const SKIP_EXTENSION = args.has('--skip-extension') || process.env.ABBENAY_SKIP_EXTENSION === '1';
 const CODE_INSTALL = args.has('--code-install');
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply build.js:49]: Added --skip-extension to the usage comment block at the top of build.js.

File: build.js [hunk 15]
```diff
@@ -367,7 +368,11 @@ async function main() {
     // Stage 3b removed: vendor list is static (single "abbenay" vendor)
 
     // Stage 4: Package extension
-    packageExtension();
+    if (!SKIP_EXTENSION) {
+        packageExtension();
+    } else {
+        console.log('\nSkipping extension packaging (--skip-extension or ABBENAY_SKIP_EXTENSION=1)\n');
+    }
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply build.js:376]: Stage 6 (--code-install) now skips with a message when --skip-extension is set, instead of failing to find a .vsix.

File: docs/decisions.md [hunk 16]
```diff
@@ -345,3 +345,23 @@ the `consumers` section is present in config (same pattern as DR-024).
...
+**Rationale:** Corporate Vertex AI proxies (e.g., Red Hat APIcast) front the
+Vertex Anthropic API but authenticate with a simple Bearer token instead of
+Google OAuth. The `@ai-sdk/google-vertex` Node.js variant requires
+`google-auth-library` and always attempts Google credential resolution, which
+throws when real GCP credentials are absent. Injecting a synthetic `authClient`
+satisfies the library's contract without dummy credentials or custom fetch
+wrappers, keeping the integration clean and forward-compatible with SDK updates.
+The `base_url` config field carries the full URL prefix (including any
+proxy-specific path segments); the SDK appends `/<model>:streamRawPredict`
+automatically.
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply docs/decisions.md]: Updated DR-026 wording to accurately reflect that the synthetic authClient avoids a fetch wrapper for auth, while proxy deployments use a small fetch wrapper for request sanitization and JSON→SSE adaptation.

File: docs/CONFIGURATION.md [hunk 17]
```diff
@@ -248,6 +249,58 @@ providers:
...
+```
+
+Set the environment variable to your Bearer token:
+
+```bash
+export VERTEX_ANTHROPIC_API_KEY="your-bearer-token-here"
+```
+
+The `base_url` should include the full path prefix up to `/models` (the engine
+appends `/<model-id>:streamRawPredict` automatically). When no `api_key` is
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply docs/CONFIGURATION.md]: Fixed. Changed to generic "When no API key is configured" to avoid referencing a non-existent `api_key` config field.

File: packages/daemon/src/core/engines.ts [hunk 18]
```diff
@@ -139,6 +143,147 @@ async function openaiCompatibleProvider(
...
+  try {
+    msg = JSON.parse(jsonStr) as Record<string, unknown>;
+  } catch {
+    return { ok: false, reason: 'parse-error' };
+  }
+  const content = Array.isArray(msg.content) ? msg.content : [];
+  let fullText = '';
+  for (const block of content) {
+    if (!block || typeof block !== 'object') {
+      return { ok: false, reason: 'non-text-content' };
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Fixed. Now validates that msg.content is an array before proceeding — non-message JSON (e.g. error payloads) returns parse-error instead of silently producing an empty response. Added test covering this case.

File: packages/daemon/src/core/engines.ts [hunk 19]
```diff
@@ -139,6 +143,150 @@ async function openaiCompatibleProvider(
...
+    `event: message_stop\ndata: ${JSON.stringify({ type: 'message_stop' })}\n`,
+  ];
+  return { ok: true as const, body: `${events.join('\n')}\n` };
+}
+
+/**
+ * Create model via @ai-sdk/google-vertex/anthropic (dynamically loaded).
+ *
+ * Supports two modes:
+ * - Standard Vertex AI: uses Google Cloud auth (requires GOOGLE_APPLICATION_CREDENTIALS)
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Fixed. Reworded to "uses Google Cloud Application Default Credentials (ADC)".

File: packages/daemon/src/core/engines.ts [hunk 20]
```diff
@@ -226,6 +374,13 @@ const ENGINES: Record<string, EngineInfo> = {
     supportsTools: true,
     createModel: (modelId, config) => dedicatedProvider('@ai-sdk/amazon-bedrock', 'createAmazonBedrock', config, modelId),
   },
+  'vertex-anthropic': {
+    id: 'vertex-anthropic',
+    requiresKey: false, // Google ADC by default; Bearer token optional via api_key
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply packages/daemon/src/core/engines.ts]: Fixed. Changed to "via apiKey" to match the actual config parameter.

**Review by cidrblock — COMMENTED** [review]
## Review: `feat(core): add vertex-anthropic engine for Vertex-hosted Claude models`

### What looks good

1. **Follows existing patterns well.** The new engine is a clean registry entry in `ENGINES`, consistent with how `bedrock` handles non-key auth. The custom factory `vertexAnthropicProvider()` is well-scoped.

2. **Proxy-compat helpers are well-extracted and testable.** `sanitizeVertexRequestBody()` and `convertAnthropicJsonToSse()` are exported functions with 13 focused unit tests covering edge cases (invalid JSON, non-text blocks, missing content arrays, stop_reason propagation, etc.).

3. **The discriminated union return type on `convertAnthropicJsonToSse()`** is a clean design — `{ ok: true; body: string } | { ok: false; reason: 'parse-error' | 'non-text-content' }` makes the caller's branching logic accurate.

4. **Good decision to remove `ABBENAY_INSECURE_TLS`** and recommend `NODE_EXTRA_CA_CERTS` instead. Significant security improvement during review.

5. **The `loadProviderFactory` subpath fix is a useful generic improvement.** Extracting the root package name from subpath exports (`@ai-sdk/google-vertex/anthropic` → `@ai-sdk/google-vertex`) benefits any future subpath import.

6. **Documentation is thorough.** CONFIGURATION.md has both proxy and ADC examples, the tool-calling limitation is documented, and DR-026 is properly recorded.

7. **Build changes are clean.** `--skip-extension` / `ABBENAY_SKIP_EXTENSION=1` correctly skips VSIX packaging AND prevents `--code-install` from running when extension is skipped.

### Items to address before merge

**1. Commit history should be squashed**

9 commits where 7 are iterative fixups to review feedback. This creates noisy history on main. Recommend squashing into 2–3 logical commits:
- `chore(build): add --skip-extension flag to skip VSIX packaging`
- `feat(core): add vertex-anthropic engine for Vertex-hosted Claude models` (with all fixes folded in)

**2. Dependency version pin is unusually specific**

```
"@ai-sdk/google-vertex": "^4.0.102"
```

Other AI SDK dependencies use `"^3.0.0"` or `"^2.0.0"`. Pinning to `^4.0.102` means npm must install at least `4.0.102`. Unless there's a specific API introduced in that version, this should be `"^4.0.0"` for consistency.

**3. Audit allowlist entry lacks context**

```
# Added 2026-04-06
https://github.com/advisories/GHSA-qj8w-gfj5-8c6v
```

Existing entries document the dependency chain and rationale (e.g., `# serialize-javascript via mocha via @vscode/test-cli (dev-only, no fix)`). Please add a comment describing which package/transitive dependency this advisory applies to.

**4. `console.warn` in the fetch wrapper could be noisy**

The body sanitization correctly uses `debug()`, but the non-text-content and parse-error warnings use `console.warn` directly. These could fire on every request for a misconfigured proxy. Consider either gating behind `debug()` or adding a warn-once pattern to avoid flooding logs.

### Worth noting (non-blocking)

- **Synthetic `authClient` is a narrow interface shim** — it only implements `getAccessToken` and `getRequestHeaders`. If `google-auth-library` evolves to call other methods, this will throw at runtime. Consider noting this coupling risk in DR-026 or adding a defensive fallback.

- **Single-delta SSE conversion for large responses** — `convertAnthropicJsonToSse()` concatenates all text into one `content_block_delta` event. The native Anthropic streaming API sends many small deltas. For very large responses this produces a single massive SSE event, which could hit parser size limits. Worth documenting as a known limitation alongside the tool-calling caveat.

- **`fetchModelsFromApi` early return** — the `if (engineId === 'vertex-anthropic') return []` follows the bedrock pattern but a brief comment explaining why (no standard model discovery API) would help future readers.

### Summary

Solid, well-tested PR. The core design is good, the proxy-compat approach (synthetic authClient + fetch wrapper) is pragmatic, and the documentation is thorough. Items 1–3 above should be addressed before merge; item 4 and the non-blocking notes can be follow-ups.

**Comment by cidrblock** [comment]
Hey Daniel — I pushed updates to your branch addressing the review feedback. Here's what changed:

**Code fixes:**
- Relaxed `@ai-sdk/google-vertex` version pin from `"^4.0.102"` to `"^4.0.0"` for consistency with other AI SDK deps
- Added context comment to the `.audit-allowlist` entry (serialize-javascript via mocha via @vscode/test-cli, dev-only)
- Switched `console.warn` to `debug()` in the fetch wrapper for non-text-content and parse-error cases to avoid log flooding with misconfigured proxies

**Commit history:**
- Squashed the 9 incremental commits into 2 clean conventional commits:
  - `chore(build): add --skip-extension flag to skip VSIX packaging`
  - `feat(core): add vertex-anthropic engine for Vertex-hosted Claude models`

**Validated locally:**
- `npm run lint` — 0 errors
- `npm test` — 420 tests passing (19 test files)
- `npm run ci:build` — full build passes

PR description updated to match.
